ast/check_test.go

```diff
@@ -2190,19 +2190,19 @@ p {
 	input.bar = "str"
 }`}},
 
-		{note: "document scope conflict", err: "test1.rego:9: rego_type_error: document annotation redeclared: test1.rego:3", modules: []string{`package test
-
-# METADATA
-# scope: document
-# schemas:
-# - input.foo: schema.number
-p { input.foo = 7 }
-
-# METADATA
-# scope: document
-# schemas:
-# - input.foo: schema.string
-p { input.foo = "str" }`}},
+		//		{note: "document scope conflict", err: "test1.rego:9: rego_type_error: document annotation redeclared: test1.rego:3", modules: []string{`package test
+		//
+		//# METADATA
+		//# scope: document
+		//# schemas:
+		//# - input.foo: schema.number
+		//p { input.foo = 7 }
+		//
+		//# METADATA
+		//# scope: document
+		//# schemas:
+		//# - input.foo: schema.string
+		//p { input.foo = "str" }`}},
```

Comment:
I commented out this test. Since we now build the annotation set during module parsing, this test fails during the parse phase. I guess the question is should a module with incorrect metadata should pass module parsing when say `ProcessAnnotation` is `true`?

Reply:
I've uncommented this test case. I've updated how the annotations are attached to the rule w/o building the annotation set during module parsing.